modulus/models/graphcast/graph_cast_net.py

```diff
@@ -129,9 +129,19 @@ def __init__(
         use_cugraphops_decoder: bool = False,
         do_concat_trick: bool = False,
         recompute_activation: bool = False,
+        partition_size: int = 1,
+        partition_group_name: Optional[str] = None,
```

Comment:
Currently, this is somewhat intentional actually. If no group is specified, it is just assumed that the partitioning (or in this case model parallelism) is across the global process group. 
But definitely open to suggestions here.

Reply:
Ah, okay. I would actually prefer if the default was to not use the global process group since the default `partition_size` is also 1. So the current defaults would work for a single GPU run, but not for non-distributed-graph training using DDP?

Reply:
Basically, what I am getting at is that it would be good to not require the user create a single GPU process group if they just want DDP and no model parallelism.

Reply:
Ah, now I understand. If implemented correctly, the logic should always be
- if partition_size <= 1, the code should run in the "single-gpu" path, i.e. the process group isn't actually needed
- if partition_size > 1, the graph is assumed to be partitioned across multiple GPUs and the passed process group will be used. If passed as None, the default process group will be used. However, there are usually checks in place to assert partition_size == group_size.

I'll add a comment about the validity of process groups though and the usage of the default process group otherwise.